pkg/agent/nodemanager/nodemanager.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2025 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodemanager
+
+import (
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+// NodeManager is to replace the upstream implementation:
+// https://raw.githubusercontent.com/kubernetes/kubernetes/refs/tags/v1.34.2/pkg/proxy/node.go.
+// The upstream NodeManager starts its own informerFactory, which is undesirable for Antrea as we rely on a single
+// shared informerFactory for all controllers. To avoid the overhead and duplication of running an additional
+// informerFactory, Antrea provides a lightweight replacement NodeManager implementation instead.
+type NodeManager struct {
```

Comment:
Actually we have a couple of examples of standalone _filtered_ informers: https://github.com/antrea-io/antrea/blob/c0f59a7758530f1c0c39818de69b287dcb187f2f/pkg/agent/client/endpoint_resolver.go#L93-L102

If you only want to watch *one* Node, I think it's acceptable to use a dedicated filtered informer, instead of using the existing shared informer factory. So if this was the only issue, you could use the upstream `NodeManager` code. However, there seems to be a key difference, which is that the upstream code panics / exits on NodeIP change. I am not sure this is the behavior we want for Antrea, or at least it requires more discussion. That behavior doesn't seem to be customizable in the upstream code.